### ISSUE TITLE
gem update mdl fails on macOS 10.14 due to mixlib-cli Ruby version requirements

### ISSUE THREAD

**Issue by jamesquilty** [markdownlint/markdownlint#289]
## Description

I wanted to update to version 0.8.0 on so I entered `gem update --user-install --no-document mdl` on the command line.

### Environment

macOS 10.14 Mojave, which has ruby 2.3.7p456 as the system-installed version.

**MDL Version:** 0.5.0

### Expected Behavior

The update reports `Successfully installed mdl 0.8.0`

### Actual Behavior

The update fails with an error.

```shell
$ gem update --user-install --no-document mdl
Updating installed gems
Updating mdl
Successfully installed kramdown-2.1.0
Fetching: kramdown-parser-gfm-1.1.0.gem (100%)
Successfully installed kramdown-parser-gfm-1.1.0
Fetching: mixlib-cli-2.1.1.gem (100%)
ERROR:  Error installing mdl:
	mixlib-cli requires Ruby version >= 2.4.
Gems updated: kramdown kramdown-parser-gfm mixlib-cli
Gems already up-to-date: mdl
$ mdl --version
0.5.0
```

## Replication Case

Any Mac running macOS 10.14 with mdl 0.5.0 already installed.

**Comment by jaymzh** [comment]
You'll need to use something like `rvm` or `rbenv`. As you can see in this line:

```
	mixlib-cli requires Ruby version >= 2.4.
```

We require at least Ruby 2.4.

**Comment by jamesquilty** [comment]
Yes, I noted that, but I'm none too keen to learn how to use `rvm` and/or `rbenv` and maintain additional Ruby versions just to use `mdl`. It also places the same burden on those who work with me. Is there no way to make version 0.8.0 function with whatever version of `mixlib-cli` works with the macOS system-installed Ruby 2.3.7?

Alternately, is the macOS 10.15 system-installed Ruby version 2.4 or later?

**Comment by jaymzh** [comment]
Since [support for ruby 2.3 has ended](https://www.ruby-lang.org/en/news/2019/03/31/support-of-ruby-2-3-has-ended), we have dropped support for `2.3`.

To answer your question; yes, if you upgrade to macOS 10.15 the system ruby is `2.6` which will work. However, it appears that future versions of macOS will not include the scripting runtimes such as `python`, `ruby`, or `perl` (search for `scripting runtimes` in [the catalina release notes](https://developer.apple.com/documentation/macos_release_notes/macos_catalina_10_15_release_notes)) and you may be interested in using [homebrew to install ruby](https://brew.sh/) or one of the aforementioned ruby environment managers such as [`rvm`](https://rvm.io/) or [`rbenv`](https://github.com/rbenv/rbenv).

**Comment by jamesquilty** [comment]
My only need for Ruby is Markdownlint, so I installed via Homebrew. The following worked for me

```shell
brew install ruby
usr/local/opt/ruby/bin/gem install --user-install --no-document mdl
emacs ~/.bash_profile  # Change PATH to point to '~/.gem/ruby/2.6.0/bin' rather than 2.3.0.
emacs ~/.gem/ruby/2.6.0/gems/mdl-0.8.0/lib/mdl/rules.rb  # Apply patch from PR #284.
```

but I had to repeat it on three different machines :-(

I think this issue can be closed.
